core/src/main/scala/kafka/server/KafkaRequestHandler.scala

```diff
@@ -69,12 +73,18 @@ class KafkaRequestHandler(id: Int,
               Exit.exit(e.statusCode)
             case e: Throwable => error("Exception when handling request", e)
           } finally {
-              request.releaseBuffer()
+            request.releaseBuffer()
           }
 
         case null => // continue
       }
     }
+    if (stopped)
```

Comment:
Could we have hit this point if `stopped` is false?

Reply:
Doesn't look like it can, removed the `if`